hazelcast/include/hazelcast/client/topic/MessageListener.h

```diff
@@ -28,27 +29,50 @@
 namespace hazelcast {
     namespace client {
         namespace topic {
+            class Message;
+            namespace impl {
+                class TopicEventHandlerImpl;
+            }
+
             /**
-             * Invoked when a message is received for the added topic. Note that topic guarantees message ordering.
-             * Therefore there is only one thread invoking onMessage. The user should not keep the thread busy, but preferably
-             * should dispatch it via an Executor. This will increase the performance of the topic.
-             *
-             * @param message the message that is received for the added topic
+             * Listen to messages from an ITopic
+             * \see ITopic::addMessageListener
              */
-            class HAZELCAST_API MessageListener {
+            class HAZELCAST_API MessageListener final {
```

Comment:
better naming suggestion: MessageListener -> listener

Reply:
listener or Listener ?

Reply:
You can do `Listener` for now but I want to change all the code base to use all lower case, similar to the stl lib and boost lib naming.